internal/k8s/controller.go

```diff
@@ -1034,6 +1034,9 @@ func validateAviConfigMap(obj interface{}) (*corev1.ConfigMap, bool) {
 
 func validateAviSecret(secret *corev1.Secret) bool {
 	if secret.Namespace == utils.GetAKONamespace() && secret.Name == lib.AviSecret {
+		// if the secret is updated or deleted we shutdown API server
```

Comment:
Though currently we are calling validateAviSecret() when secret gets added/updated, one can potentially call function from anywhere, is this the right place to trigger a shutdown ?

Reply:
This is on delete/update secret, specifically for `avi-secret`. do you see any issues around calling this from the event handler itself?

Reply:
Yeah, we can trigger a shutdown for the event handler.